library/Zend/InputFilter/Input.php

```diff
@@ -326,6 +326,10 @@ public function isValid($context = null)
             return true;
         }
 
+        if ($empty && !$this->isRequired()) {
```

Comment:
And `$empty` should be `$empty === true`? :wink: I see no valid reason for comparing a method, which returns a boolean value by definition, to `true`/`false`, but perhaps you can provide a use case for that?